### PR TITLE
Implement linux-i2c vesion of I2C library for use in Ubuntu platform

### PR DESCRIPTION
**Pull request by pimvanpelt**
This adds `i2c.dev_no` to mos.yml, which instructs the driver to open /dev/i2c-? as backend for I2C.
It then implements a Linux I2C version of the driver, using ioctl() to select the slave, and read/write to write to it.

Notes:
*    Linux doesn't have a call to free/reset the bus
*    Linux doesn't expose the underlying SCL/SDA pins, so bitbanging a reset is not possible.
*    Linux doesn't allow stop bit to be absent (!)
*    Linux only supports 100KHz frequency.

Basic functionality works -- reading and writing, SMBUS register read/write, and scanning.

### CODE REVIEW

File: src/ubuntu/ubuntu_i2c_master.c [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  int freq;
+  int read_timeout_ms;
+  int fd;
+  int dev_no;
+};
+
+static const char *mgos_i2c_dev_filename(struct mgos_i2c *c) {
+  static char fn[256];
+
+  if (!c) {
```
Comment (rojer) [inline src/ubuntu/ubuntu_i2c_master.c]: style point: we don't use boolean comparison for pointers, so - `if (c == NULL)` please

File: src/ubuntu/ubuntu_i2c_master.c [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  }
+  LOG(LL_ERROR, ("Invalid I2C settings"));
+  return NULL;
+}
+
+void mgos_i2c_close(struct mgos_i2c *c) {
+  if (!c) {
+    return;
+  }
+  if (c->fd > 0) {
```
Comment (rojer) [inline src/ubuntu/ubuntu_i2c_master.c]: strictly speaking, valid fd can be zero. so use -1 as invalid value and change this to >=

File: src/ubuntu/ubuntu_i2c_master.c (hunk 1 above)
Comment (pimvanpelt) [reply src/ubuntu/ubuntu_i2c_master.c]: Done. I will have to unlearn this :) and go over the other code I submitted which consistently uses `if (!c)`. Updated here.

File: src/ubuntu/ubuntu_i2c_master.c (hunk 2 above)
Comment (pimvanpelt) [reply src/ubuntu/ubuntu_i2c_master.c]: Good catch. Will also update the platform code, as I consistently have used >0 there.

**Comment by rojer** [comment]
https://github.com/cesanta/mos-libs/commit/d2806f768ea9742c036fb3214499f7958adc1695

**Pull request closed without merging**